src/util.cpp

```diff
@@ -158,18 +157,13 @@ bool LockDirectory(const fs::path& directory, const std::string lockfile_name, b
     // Create empty lock file if it doesn't exist.
     FILE* file = fsbridge::fopen(pathLockFile, "a");
     if (file) fclose(file);
-
-    try {
-        auto lock = MakeUnique<boost::interprocess::file_lock>(pathLockFile.string().c_str());
-        if (!lock->try_lock()) {
-            return false;
-        }
-        if (!probe_only) {
-            // Lock successful and we're not just probing, put it into the map
-            dir_locks.emplace(pathLockFile.string(), std::move(lock));
-        }
-    } catch (const boost::interprocess::interprocess_exception& e) {
-        return error("Error while attempting to lock directory %s: %s", directory.string(), e.what());
+    auto lock = MakeUnique<fsbridge::FileLock>(pathLockFile);
+    if (!lock->TryLock()) {
+        return error("Error while attempting to lock directory %s: %s", directory.string(), lock->GetReason());
```

Comment:
For reference, how these are logged by the callers:
https://github.com/bitcoin/bitcoin/blob/2b67354aa584c4aabae049a67767ac7b70e2d01a/src/init.cpp#L1183-L1185

https://github.com/bitcoin/bitcoin/blob/2b67354aa584c4aabae049a67767ac7b70e2d01a/src/wallet/db.cpp#L136-L139